### PR TITLE
Integrate fork choice tests

### LINKED ISSUE #2162: Integrate v0.12.1 fork choice tests
### Description
Had to revert https://github.com/PegaSysEng/teku/pull/2154 on the 0.12.1-integration branch because those are the 0.11.1 fork choice tests.  We need to get the v0.12 fork choice tests and integrate them.  

Need to also fix the way they're integrated as they're being run as a separate test, not actually picked up by the reference test runner because they're under `src/test/java` instead of `src/test/resources` with the rest of the reference tests.

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
Re-integrate fork choice tests that are now compatible with spec version 0.12.2.

## Fixed Issue(s)

Fixes #2162

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: eth-reference-tests/src/referenceTest/resources/fork_choice_tests/cache/attestation_0.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+aggregation_bits: '0x11'
```
Comment (mbaxter) [inline eth-reference-tests/src/referenceTest/resources/fork_choice_tests/cache/attestation_0.yaml]: Looks like you've added these files to `fork-choice-tests` resources and `eth-reference-tests` resources.  Should these ref-test files be deleted?

Comment (cemozerr) [reply eth-reference-tests/src/referenceTest/resources/fork_choice_tests/cache/attestation_0.yaml]: Yep, the ones in `eth-reference-tests` should be deleted. Good catch.

Comment (cemozerr) [reply eth-reference-tests/src/referenceTest/resources/fork_choice_tests/cache/attestation_0.yaml]: Done.

File: eth-reference-tests/build.gradle [hunk 2]
```diff
@@ -2,13 +2,19 @@ jar { enabled = false }
...
   referenceTestImplementation project(':ethereum:datastructures')
+  referenceTestImplementation project(':ethereum:statetransition')
   referenceTestImplementation project(':eth-tests')
+  referenceTestImplementation project(':protoarray')
   referenceTestImplementation project(':ssz')
   referenceTestImplementation project(':storage')
+  referenceTestImplementation project(':infrastructure:async')
   referenceTestImplementation project(':util')
 
+  referenceTestImplementation testFixtures(project(':storage'))
```
Comment (mbaxter) [inline eth-reference-tests/build.gradle]: Looks like these changes to the `eth-reference-tests` dependencies are no longer necessary either.

**Review by mbaxter — APPROVED** [review]
LGTM

File: eth-reference-tests/build.gradle (hunk 2 above)
Comment (cemozerr) [reply eth-reference-tests/build.gradle]: Removed.

**Pull request merged**